lib/queue.js

```diff
@@ -812,6 +821,39 @@ Queue.prototype.clean = function (grace, type) {
   });
 };
 
+/**
+ * Returns a promise that resolves when active jobs are cleared
+ *
+ * @returns {Promise}
+ */
+Queue.prototype.whenCurrentJobsFinished = function(){
+  var resolve, reject;
+  var promise = new Promise(function(innerResolve, innerReject) {
+    resolve = innerResolve;
+    reject = innerReject;
+  });
+
+  var _this = this;
+
+  var resolver = function(count) {
+    if(count === 0) {
+      resolve();
+    }
+    return count;
+  };
+
+  this.getActiveCount().then(resolver).
+    then(function(count) {
+      if(count === 0) return null;
+      _this.on('completed', function() {
+        _this.getActiveCount().then(resolver).catch(reject);
+      });
+      return null;
+    }).catch(reject);
```

Comment:
I think this function would be a little easier to read, and no need for a deferred, if this logic was inside the top promise:

```js
var _this = this;
return new Promise(function(resolve, reject) {
  _this.getActiveCount().then(function(count) {
    if (count === 0) resolve();
    else {
      _this.on('completed', function() {
        _this.getActiveCount().then(function(count) {
          if (count === 0) resolve();
        }).catch(reject);
      });
    }
  }).catch(reject);
});
```

also, if we're only considering "current" jobs / the queue is paused, then I don't believe we need to call `_this.getActiveCount()` multiple times:

```js
var _this = this;
return new Promise(function(resolve, reject) {
  _this.getActiveCount().then(function(count) {
    if (count === 0) resolve();
    else {
      _this.on('completed', _.after(count, resolve));
    }
  }).catch(reject);
});
```